x-pack/plugins/security_solution/server/endpoint/routes/metadata/support/unenroll.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { APICaller, IScopedClusterClient } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { metadataMirrorIndexPattern } from '../../../../../common/endpoint/constants';
+import { EndpointStatus } from '../../../../../common/endpoint/types';
+
+const KEEPALIVE = '10s';
+
+export interface HostId {
+  host: {
+    id: string;
+  };
+}
+
+interface HitSource {
+  _source: HostId;
+}
+
+export async function findUnenrolledHostByHostId(
+  client: IScopedClusterClient,
+  hostId: string
+): Promise<HostId | undefined> {
+  const queryParams = {
+    index: metadataMirrorIndexPattern,
+    body: {
+      size: 1,
+      _source: ['host.id'],
+      query: {
+        bool: {
+          must: {
+            term: { 'host.id': hostId },
```

Comment:
I think we can move this down to the filter context since it should result in `yes/no` rather than a score for relevance